lib/ansible/modules/system/timezone.py

```diff
@@ -99,8 +100,14 @@ def __new__(cls, module):
         """
         if get_platform() == 'Linux':
             timedatectl = module.get_bin_path('timedatectl')
-            if timedatectl is not None and module.run_command(timedatectl)[0] == 0:
-                return super(Timezone, SystemdTimezone).__new__(SystemdTimezone)
+            if timedatectl is not None:
+                rc, stdout, stderr = module.run_command(timedatectl)
+                if rc == 0:
+                    return super(Timezone, SystemdTimezone).__new__(SystemdTimezone)
+                elif any(module.get_bin_path(cmd) is not None for cmd in ('dpkg-reconfigure', 'tzdata-update')):
+                    return super(Timezone, NosystemdTimezone).__new__(NosystemdTimezone)
+                else:
```

Comment:
I dont think this change is needed, it would just return NonSystemd by default, no need to search for the other commands, that happens later already in NonSystemdTimezone class

Reply:
This is to give better error message.

If the system **has** `timedatectl` command and doesn't have neither `dpkg-reconfigure` nor `tzdata-update`, the timezone should be set using `timedatectl`.
In such case, an error coming from `NonSytemdTimezone` is very confusing and users don't know how to fix it (the error says 'cannot read configuration file "/etc/sysconfig/clock" for name') .
So IMO, it'd better to clarify in the error message that the problem happens from the `timedatectl` command.

Reply:
I would say that the 'cannot read' is an incomplete error message .. 'tried to configure using X and but failed as I cannot read Y' seems a better error message for both cases. 

if you want to add an additional warning 'timedatectl found but not usable, falling back to other methods', it might help the user diagnose the issue on failure.